src/media/js/site/login.js

```diff
@@ -14,8 +14,39 @@ export class DefaultUnauthorizedHandler extends React.Component {
 };
 
 
+export class DefaultTOSSignatureHandler extends React.Component {
```

Comment:
They're all sort of related in that we require login, permission, and TOS signature before we allow access to the tools, and show a roadblock handler in the event that any of those conditions aren't met.